src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -394,7 +398,10 @@ internal static class EventLogEntryCollectionExtensions
     {
         internal static EventLogEntry LastOrDefault(this EventLogEntryCollection elec)
         {
-            return elec.Count > 0 ? elec[elec.Count - 1] : null;
+            EventLogEntry eventLogEntry = null;
+            eventLogEntry = Helpers.RetrieveEntryOrMessage<EventLogEntry>(() => elec.Count > 0 ? elec[elec.Count - 1] : null);
```

Comment:
In many cases the compiler can infer the generic type constraint. eg in this case you can probably just write
```c#
{
return Helpers.RetrieveEntryOrMessage(() => elec.Count > 0 ? elec[elec.Count - 1] : null);
}
```